module/synchronization/core.go

```diff
@@ -300,7 +358,6 @@ func (c *Core) getRequestableItems() ([]uint64, []flow.Identifier) {
 	// create list of all the block IDs blocks that are missing
 	var blockIDs []flow.Identifier
 	for blockID, status := range c.blockIDs {
```

Comment:
One thing I'm concerned with this `range` is that:
When a node is falling far behind, it will request both by height and by blockIDs, and this blockIDs will keep growing because it takes time for the finalized height to go up to prune it. This will  cause `range c.blockIDs` take longer and longer.

For instance, if finalized height is 1000, and it was asked to fetch blocks from [1000000-2000000], then `c.blockIDs` will have 1000000 items that won't be pruned until the finalization has caught up.